silence/silence.go

```diff
@@ -276,6 +325,11 @@ func (s *Silences) setSilence(sil *pb.Silence) error {
 	if err != nil {
 		return err
 	}
+
+	_, err = s.mc.Add(sil)
+	if err != nil {
+		return err
+	}
```

Comment:
Ah, I see you basically circumvent this by lazily adding the silence to the cache when it doesn't exist yet?